src/StargateEventListener/Handlers/StargateEventHandlers.ts

```diff
@@ -0,0 +1,269 @@
+            debug(ERROR_EVENTS.STARGATE, `No NodeManagement contract address found for network: ${network.type}`)
+            return
+        }
+
+        const nodeDelegatedEvents = await parseNodeDelegatedEvents(beat, nodeManagementAddress, thor)
+
+        if (nodeDelegatedEvents.length > 0) {
+            debug(
+                ERROR_EVENTS.STARGATE,
+                `Found ${nodeDelegatedEvents.length} NodeDelegated events in beat ${beat.number}`,
+            )
+
+            // Process each event
+            for (const eventData of nodeDelegatedEvents) {
+                await processNodeDelegatedEvent(eventData, refetchStargateData)
+            }
+        }
+    } catch (err) {
+        error(ERROR_EVENTS.STARGATE, "Error handling NodeDelegated event:", err)
+    }
+}
+
+/**
+ * Parse NodeDelegated events from a beat
+ */
+const parseNodeDelegatedEvents = async (
+    beat: Beat,
+    nodeManagementContract: string,
+    thor: ThorClient,
+): Promise<NodeDelegatedEventData[]> => {
+    try {
+        const block = await thor.blocks.getBlockCompressed(beat.id)
+
+        if (!block?.transactions) {
+            debug(ERROR_EVENTS.STARGATE, `No transactions found in block ${beat.number}`)
+            return []
+        }
+
+        const events = await processTransactionsForEvents(block.transactions, nodeManagementContract, thor, beat)
+        debug(ERROR_EVENTS.STARGATE, `Parsed ${events.length} NodeDelegated events from beat ${beat.number}`)
+
+        return events
+    } catch (err) {
+        error(ERROR_EVENTS.STARGATE, "Error parsing NodeDelegated events:", err)
+        return []
+    }
+}
+
+/**
+ * Process all transactions in a block to find NodeDelegated events
+ */
+const processTransactionsForEvents = async (
+    transactions: string[],
+    nodeManagementContract: string,
+    thor: ThorClient,
+    beat: Beat,
+): Promise<NodeDelegatedEventData[]> => {
+    const events: NodeDelegatedEventData[] = []
+
+    for (const txId of transactions) {
+        try {
+            const transactionEvents = await processTransactionForEvents(txId, nodeManagementContract, thor, beat)
+            events.push(...transactionEvents)
+        } catch (txError) {
+            debug(ERROR_EVENTS.STARGATE, `Error processing transaction ${txId}:`, txError)
+        }
+    }
+
+    return events
+}
+
+/**
+ * Process a single transaction to find NodeDelegated events
+ */
+const processTransactionForEvents = async (
+    txId: string,
+    nodeManagementContract: string,
+    thor: ThorClient,
+    beat: Beat,
+): Promise<NodeDelegatedEventData[]> => {
+    const receipt = await thor.transactions.getTransactionReceipt(txId)
+
+    if (!receipt?.outputs) {
+        return []
+    }
+
+    const events: NodeDelegatedEventData[] = []
+
+    for (const output of receipt.outputs) {
+        if (!output.events) continue
+
+        const outputEvents = processOutputEvents(output.events, nodeManagementContract, beat, txId)
```

Comment:
To process the events, you can use a `ReceiptProcessor`
You can get the right one with `getReceiptProcessor()`